lib/wovnrb/html_replacers/link_replacer.rb

```diff
@@ -1,5 +1,12 @@
 module Wovnrb
   class LinkReplacer < ReplacerBase
+    module FileExtension
+      IMG_FILES = '(?!jp$)jpe?g?|bmp|gif|png|btif|tiff?|psd|djvu?|xif|wbmp|webp|p(n|b|g|p)m|rgb|tga|x(b|p)m|xwd|pic|ico|fh(c|4|5|7)?|xif|f(bs|px|st)'
+      AUDIO_FILES = 'mp(3|2)|m(p?2|3|p?4|pg)a|midi?|kar|rmi|web(m|a)|aif(f?|c)|w(ma|av|ax)|m(ka|3u)|sil|s3m|og(a|g)|uvv?a'
+      VIDEO_FILES = 'm(x|4)u|fl(i|v)|3g(p|2)|jp(gv|g?m)|mp(4v?|g4|(^?!$)e?g?)|m(1|2)v|ogv|m(ov|ng)|qt|uvv?(h|m|p|s|v)|dvb|mk(v|3d|s)|f4v|as(x|f)|w(m(v|x)|vx)'
```

Comment:
This pattern could not detect

- http://favy.tips/index.xvid
- http://favy.tips/index.mpeg

Btw, maintenance this complex pattern is very painful imo :|

Reply:
Ah, my bad! I put an extra `^` in the regex for video files and mpeg is not detected.

True, it was painful to write and check too. We should decide on a set of supported file extensions for each type to maintain consistency.

Reply:
I think so too.

Why don't you check extension name only?
```
require 'uri'
require 'set'

ext = File.extname(URI.parse("http://日本語ドメイン.jp/path/a.jpg?path=a.html&path2=/dir/b.txt".gsub(/[[:^ascii:]]+/, 'zzz')).path)
# ".jpg"
ignore_exts = Set.new(%w[.jpg .png]) # you can add all extensions
# #<Set: {".jpg", ".png"}>
ignore_exts.include?(ext)
# true
```

The example replace non-ascii to `zzz`.
Because URI.parse is fail if url include non-ascii code. 

This is just suggestion.
If you have a good idea, you can ignore it.

Reply:
Let's just fix the issue and keep these patterns as they're now ;)